### ISSUE TITLE
[Question] How to control whether 2.0 or 1.x template is used

### ISSUE THREAD

**Issue by lambdakris** [dotnet/templating#653]
This is a silly thing but I can't seem to pin down how the 2.0 vs 1.x templates are selected by new3. It doesn't seem to be just the --framework switch... or is it?

**Comment by mlorbetske** [comment]
Good question. Templates with a common groupIdentity show as one & each template in each group identity can have options and defaults. If a group identity is used by more than one template, then when resolution for invocation happens (ex. calling up help or trying to create an instance of the template), the specified parameters are used to try to narrow down to exactly one template the user might have meant. This, usually doesn't get us all the way there (ex. dotnet new3 classlib resolves to both its 1.x and 2.0 variants), so we indicate which is the more preferred in the event of ambiguity with the "precedence" property - in a group, the template with the highest precedence will be selected & it's defaults applied.

Summing up, the parameters are first used to narrow down the set of templates (the templates for which the paramerers are valid are selected), they're ordered by precedence (descending) and then the first one is picked.

**Comment by mlorbetske** [comment]
Closing this, feel free to re-open if there's more info needed
